--- src/original/moba-desktop/src/windows/game-overlay/game-overlay.window.js
+++ src/modified/moba-desktop/src/windows/game-overlay/game-overlay.window.js
@@ -50,6 +50,7 @@
             preload: path.join(__dirname, 'game-overlay.preload.js'),
             nodeIntegration: false,
             contextIsolation: false,
+            enableRemoteModule: true,
             webSecurity: true,
             allowRunningInsecureContent: false,
             experimentalFeatures: false,
