diff --color -raN -U 6 original/moba-desktop/src/app/app.js modified/moba-desktop/src/app/app.js
--- original/moba-desktop/src/app/app.js	2022-12-06 21:34:33.379586165 -0300
+++ modified/moba-desktop/src/app/app.js	2022-12-01 00:21:20.316659763 -0300
@@ -70,13 +70,12 @@
         ipc_protocol_1.init();
         app_internal_webview_1.init();
         valorant_lcu_controller_1.init();
         main_utils_1.showOrCreateWindow();
         app_tray_1.init();
         deep_linking_1.addDeepLinkingListeners();
-        setInterval(promo_1.maybeShowSplashScreen, 1000 * 60 * 10);
     }));
     electron_1.app.on('window-all-closed', () => {
     });
     electron_1.app.on('before-quit', () => {
         var _a;
         const lolSettingsFetcher = lol_settings_fetcher_1.LolSettingsFetcher.getInstance();
diff --color -raN -U 6 original/moba-desktop/src/app/deep-linking.js modified/moba-desktop/src/app/deep-linking.js
--- original/moba-desktop/src/app/deep-linking.js	2022-12-06 21:34:33.379586165 -0300
+++ modified/moba-desktop/src/app/deep-linking.js	2022-12-01 00:16:42.428007645 -0300
@@ -34,12 +34,17 @@
 const main_utils_1 = require("../windows/main/main.utils");
 function initDeepLinking() {
     electron_1.app.removeAsDefaultProtocolClient(app_config_1.MOBA_PROTOCOL_NAME);
     if (environment_1.IS_ELECTRON_DEV && environment_1.isWin) {
         electron_1.app.setAsDefaultProtocolClient(app_config_1.MOBA_PROTOCOL_NAME, process.execPath, [path.resolve(process.argv[1])]);
     }
+    else if (process.defaultApp){
+        if (process.argv.length >= 2) {
+            electron_1.app.setAsDefaultProtocolClient(app_config_1.MOBA_PROTOCOL_NAME, process.execPath, [path.resolve(process.argv[1])]);
+        }
+    }
     else {
         electron_1.app.setAsDefaultProtocolClient(app_config_1.MOBA_PROTOCOL_NAME);
     }
 }
 exports.initDeepLinking = initDeepLinking;
 function addDeepLinkingListeners() {
@@ -49,13 +54,13 @@
             processDeepLinkingUrl(url);
         }
         logs_1.appUrlsLog.debug(`open-url: ${url}`);
         main_utils_1.showAndRestoreMainWindow();
     });
     electron_1.app.on('second-instance', (_e, argv) => {
-        if (environment_1.isWin) {
+        if (environment_1.isWin || environment_1.isLinux) {
             const url = argv.find(it => it.startsWith(`${app_config_1.MOBA_PROTOCOL_NAME}:`));
             url && processDeepLinkingUrl(url);
         }
         logs_1.appUrlsLog.debug('second-instance: ', argv);
         main_utils_1.showAndRestoreMainWindow();
     });
diff --color -raN -U 6 original/moba-desktop/src/dialogs/select-game-folder-dialog.js modified/moba-desktop/src/dialogs/select-game-folder-dialog.js
--- original/moba-desktop/src/dialogs/select-game-folder-dialog.js	2022-12-06 21:34:33.376586114 -0300
+++ modified/moba-desktop/src/dialogs/select-game-folder-dialog.js	2022-11-25 13:45:52.402640859 -0300
@@ -10,18 +10,31 @@
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.showSelectLolClientFolderDialog = void 0;
 const ipc_messages_types_1 = require("../../../moba-shared/src/common/types/desktop/ipc-messages.types");
 const remote_1 = require("@electron/remote");
 const ipc_renderer_1 = require("../utils/ipc-renderer");
+const environment_1 = require("../utils/environment");
 function showSelectLolClientFolderDialog(window) {
     if (remote_1.dialog) {
+        let install_message;
+        let install_title;
+        if (environment_1.isLinux){
+            install_message = 'Choose the LoL installation path. e.g $WINEPREFIX/drive_c/Riot Games/League of Legends/';
+            install_title = 'Choose the LoL installation path. e.g $WINEPREFIX/drive_c/Riot Games/League of Legends/';
+        }
+        else
+        {
+            install_message = 'Choose the LoL installation path. e.g /Applications/League of Legends.app/';
+            install_title = 'Choose the LoL installation path. e.g C:\\Riot Games\\League of Legends';
+        }
+            
         return remote_1.dialog
             .showOpenDialog(window, {
-            title: 'Choose the LoL installation path. e.g C:\\Riot Games\\League of Legends',
-            message: 'Choose the LoL installation path. e.g /Applications/League of Legends.app/',
+            title: install_title,
+            message: install_message,
             buttonLabel: 'Select Directory',
             properties: ['openDirectory', 'treatPackageAsDirectory', 'showHiddenFiles'],
         })
             .then((result) => __awaiter(this, void 0, void 0, function* () {
             if (!result.canceled) {
                 const folder = result.filePaths ? result.filePaths[0] : null;
diff --color -raN -U 6 original/moba-desktop/src/lol/utils/in-game-imports/in-game-items-import.js modified/moba-desktop/src/lol/utils/in-game-imports/in-game-items-import.js
--- original/moba-desktop/src/lol/utils/in-game-imports/in-game-items-import.js	2022-12-06 21:34:33.316585081 -0300
+++ modified/moba-desktop/src/lol/utils/in-game-imports/in-game-items-import.js	2022-11-25 13:47:38.344563355 -0300
@@ -35,19 +35,24 @@
 const path_1 = __importDefault(require("path"));
 const fse = __importStar(require("fs-extra"));
 const logs_1 = require("../../../utils/logs");
 const in_game_imports_types_1 = require("../../../../../moba-shared/src/lol/types/in-game-imports.types");
 const lcu_connection_1 = require("../../../utils/lcu-connection");
 const root_store_1 = require("../../../common/store/root-store");
+const environment_1 = require("../../../utils/environment");
 function importLolGameItems(gameItemsData) {
     return __awaiter(this, void 0, void 0, function* () {
         try {
             if (!gameItemsData.blocks.length)
                 return in_game_imports_types_1.LcuGameItemsImportStatus.EMPTY_BLOCKS_LIST;
             const processGamePath = yield lcu_connection_1.getLCUPathFromProcess();
-            const gamePath = processGamePath || root_store_1.getAppSettings().getGameFolderPath();
+            let gamePath;
+            if(environment_1.isLinux)
+                gamePath = root_store_1.getAppSettings().getGameFolderPath();
+            else
+                gamePath = processGamePath || root_store_1.getAppSettings().getGameFolderPath();
             if (!gamePath)
                 return in_game_imports_types_1.LcuGameItemsImportStatus.GAME_PATH_NOT_FOUND;
             const filePath = path_1.default.join(gamePath, 'Config', 'Global', 'Recommended', '!mobalytics.json');
             yield fse.ensureFile(filePath);
             const blocks = gameItemsData.blocks.map(it => (Object.assign(Object.assign({}, it), { recMath: false, minSummonerLevel: -1, maxSummonerLevel: -1, showIfSummonerSpell: '', hideIfSummonerSpell: '' })));
             const content = {
diff --color -raN -U 6 original/moba-desktop/src/utils/desktop-bridge.js modified/moba-desktop/src/utils/desktop-bridge.js
--- original/moba-desktop/src/utils/desktop-bridge.js	2022-12-06 21:34:33.301584822 -0300
+++ modified/moba-desktop/src/utils/desktop-bridge.js	2022-12-01 00:33:35.182704044 -0300
@@ -29,14 +29,15 @@
 const show_file_dialog_1 = require("../dialogs/show-file-dialog");
 const version_const_1 = require("./version.const");
 const platform = os.platform();
 exports.bridge = {
     version: remote_1.app.getVersion() === '0.0.1' ? version_const_1.DEV_DESKTOP_APP_VERSION : remote_1.app.getVersion(),
     platform,
-    isWin: platform !== 'darwin',
+    isWin: platform === 'win32',
     isMac: platform === 'darwin',
+    isLinux: platform === 'linux',
     logsPath: path.resolve(remote_1.app.getPath('logs')),
     settings: null,
     ipcRendererSend: ipc_renderer_1.ipcRendererSend,
     ipcRendererOn: ipc_renderer_1.ipcRendererOn,
     ipcRendererInvoke: ipc_renderer_1.ipcRendererInvoke,
     forceShowCurrentWindow: () => {
diff --color -raN -U 6 original/moba-desktop/src/utils/environment.js modified/moba-desktop/src/utils/environment.js
--- original/moba-desktop/src/utils/environment.js	2022-12-06 21:34:33.301584822 -0300
+++ modified/moba-desktop/src/utils/environment.js	2022-11-25 13:54:08.907827359 -0300
@@ -21,18 +21,19 @@
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.AppExecPath = exports.LolLogsSubPath = exports.RiotClientName = exports.RiotClientNameMac = exports.RiotClientNameWin = exports.LolProcessName = exports.LoLProcessNameMac = exports.LoLProcessNameWin = exports.IS_ELECTRON_DEV = exports.IS_JEST_TESTING = exports.isX64 = exports.isWin = exports.isOsX = exports.APP_USER_MODEL_ID = void 0;
 const path = __importStar(require("path"));
 exports.APP_USER_MODEL_ID = 'MobalyticsHQ.DesktopApp';
 exports.isOsX = process.platform === 'darwin';
 exports.isWin = process.platform === 'win32';
+exports.isLinux = process.platform === 'linux';
 exports.isX64 = process.arch === 'x64';
 exports.IS_JEST_TESTING = process.env.JEST_WORKER_ID !== undefined;
-exports.IS_ELECTRON_DEV = require('electron-is-dev');
+exports.IS_ELECTRON_DEV = false;
 exports.LoLProcessNameWin = 'LeagueClient.exe';
 exports.LoLProcessNameMac = 'LeagueClient.app';
-exports.LolProcessName = exports.isWin ? exports.LoLProcessNameWin : exports.LoLProcessNameMac;
+exports.LolProcessName = (exports.isWin || exports.isLinux) ? exports.LoLProcessNameWin : exports.LoLProcessNameMac;
 exports.RiotClientNameWin = 'RiotClientServices.exe';
 exports.RiotClientNameMac = '';
 exports.RiotClientName = exports.isWin ? exports.RiotClientNameWin : exports.RiotClientNameMac;
 exports.LolLogsSubPath = path.join('Logs', 'GameLogs');
 exports.AppExecPath = `"${path.normalize(path.resolve(process.execPath))}"`;
 //# sourceMappingURL=environment.js.map
\ No hay ningún carácter de nueva línea al final del fichero
diff --color -raN -U 6 original/moba-desktop/src/windows/main/main.window.js modified/moba-desktop/src/windows/main/main.window.js
--- original/moba-desktop/src/windows/main/main.window.js	2022-12-06 21:34:33.300584804 -0300
+++ modified/moba-desktop/src/windows/main/main.window.js	2022-11-25 13:54:08.907827359 -0300
@@ -31,35 +31,47 @@
 const navigation_1 = require("../../utils/navigation");
 const window_unresponsive_dialog_1 = require("../../dialogs/window-unresponsive-dialog");
 const remote = __importStar(require("@electron/remote/main"));
 const root_store_1 = require("../../common/store/root-store");
 const lang_1 = require("../../../../moba-shared/src/common/types/lang");
 const url_1 = require("../../../../moba-shared/src/common/types/url");
+const adblocker1 = require("@cliqz/adblocker-electron");
+const fs1 = require("fs");
 let mainWindow;
 let appWindowState;
 function getMainWindowState() {
     return appWindowState;
 }
 exports.getMainWindowState = getMainWindowState;
 function getMainWindow() {
     return mainWindow;
 }
 exports.getMainWindow = getMainWindow;
-function ensureMainWindow() {
+async function ensureMainWindow() {
     logs_1.generalLog.debug('Ensure main window');
     if (!mainWindow) {
         const { width: settingsWidth, height: settingsHeight } = root_store_1.getAppSettings().getDefaultAppWindowSize();
         const { width: fallbackWidth, height: fallbackHeight } = electron_1.screen.getPrimaryDisplay().workAreaSize;
         const defaultWidth = Math.min(lang_1.isDef(settingsWidth) ? settingsWidth : fallbackWidth, fallbackWidth);
         const defaultHeight = Math.min(lang_1.isDef(settingsHeight) ? settingsHeight : fallbackHeight, fallbackHeight);
         appWindowState = electron_window_state_1.default({
             defaultWidth,
             defaultHeight,
             fullScreen: false,
             maximize: false,
         });
+        const blocker = await adblocker1.ElectronBlocker.fromLists(fetch,adblocker1.fullLists,
+                {
+                  enableCompression: true,
+                },
+                {
+                  path: path.default.join(electron_1.app.getPath('userData'),"adblock.bin"),
+                  read: async (...args) => fs1.readFileSync(...args),
+                  write: async (...args) => fs1.writeFileSync(...args),
+                },
+              );
         mainWindow = new electron_1.BrowserWindow({
             minWidth: desktop_default_settings_1.MIN_WINDOW_WIDTH,
             minHeight: desktop_default_settings_1.MIN_WINDOW_HEIGHT,
             title: desktop_default_settings_1.APP_TITLE,
             x: appWindowState.x,
             y: appWindowState.y,
@@ -81,12 +93,13 @@
             frame: false,
             titleBarStyle: 'hidden',
             maximizable: true,
         });
         appWindowState.manage(mainWindow);
         remote.enable(mainWindow.webContents);
+        blocker.enableBlockingInSession(mainWindow.webContents.session);
         mainWindow.loadURL(navigation_1.getSafetyWindowURL(url_1.AppPath.ONBOARDING_ROOT));
         navigation_1.initNavigationSecurity(mainWindow);
         mainWindow.on('ready-to-show', () => {
             if (mainWindow) {
                 mainWindow.show();
                 mainWindow.focus();