diff -uraN src/original/moba-desktop/src/windows/main/main.window.js src/modified/moba-desktop/src/windows/main/main.window.js
--- src/original/moba-desktop/src/windows/main/main.window.js	2021-11-03 20:17:51.312039322 +0000
+++ src/modified/moba-desktop/src/windows/main/main.window.js	2021-11-03 20:37:06.735329615 +0000
@@ -35,6 +35,7 @@
 const ipc_messages_types_1 = require("../../../../moba-web/src/common/types/desktop/ipc-messages.types");
 const url_1 = require("../../../../moba-web/src/common/types/url");
 const auto_update_1 = require("../../app/auto-update");
+const app_config_1 = require("../../../../moba-web/src/app.config");
 let mainWindow;
 let appWindowState;
 function getMainWindowState() {
@@ -54,6 +55,25 @@
             fullScreen: false,
             maximize: false,
         });
+        electron_1.dialog.showMessageBox(null, {
+            title: 'Registering ' + app_config_1.MOBA_PROTOCOL_NAME,
+            buttons: ['Dismiss'],
+            type: 'info',
+            message: 'Application is not responding…',
+        });
+        const partition = `persist:${app_config_1.MOBA_PROTOCOL_NAME}`
+        const ses = electron_1.session.fromPartition(partition)
+
+        ses.protocol.registerHttpProtocol(app_config_1.MOBA_PROTOCOL_NAME, (request, callback) => {
+            electron_1.dialog.showMessageBox(null, {
+                title: 'RECIEVED ' + app_config_1.MOBA_PROTOCOL_NAME,
+                buttons: ['Dismiss'],
+                type: 'info',
+                message: 'Application is not responding…',
+            });
+            const url = request.url.substr(7)
+            callback({ path: path.normalize(`${__dirname}/${url}`) })
+        })
         mainWindow = new electron_1.BrowserWindow({
             minWidth: desktop_default_settings_1.MIN_WINDOW_WIDTH,
             minHeight: desktop_default_settings_1.MIN_WINDOW_HEIGHT,
@@ -72,6 +92,7 @@
                 allowRunningInsecureContent: false,
                 experimentalFeatures: false,
                 enableRemoteModule: true,
+                partition
             },
             show: false,
             resizable: true,
@@ -241,4 +262,4 @@
     window.loadURL(navigation_1.getSafetyWindowURL(url_1.AppPath.FORCE_DESKTOP_APP_UPDATE));
 }
 exports.redirectToForceUpdatePage = redirectToForceUpdatePage;
-//# sourceMappingURL=main.window.js.map
\ No newline at end of file
+//# sourceMappingURL=main.window.js.map
