--- src/original/moba-desktop/src/windows/about/about.window.js	2023-03-27 13:46:14.381197108 +0100
+++ src/modified/moba-desktop/src/windows/about/about.window.js	2023-03-27 13:46:14.507864600 +0100
@@ -44,6 +44,7 @@
                 contextIsolation: false,
                 webSecurity: true,
                 allowRunningInsecureContent: false,
+                enableRemoteModule: true,
                 experimentalFeatures: false,
                 autoplayPolicy: 'user-gesture-required',
             },
@@ -63,4 +64,4 @@
     aboutWindow ? (aboutWindow.isVisible() ? aboutWindow.hide() : aboutWindow.show()) : createAboutWindow();
 }
 exports.toggleAboutWindow = toggleAboutWindow;
-//# sourceMappingURL=about.window.js.map
\ No newline at end of file
+//# sourceMappingURL=about.window.js.map
