--- src/original/moba-desktop/src/dialogs/authorization-error-dialog.js
+++ src/modified/moba-desktop/src/dialogs/authorization-error-dialog.js
@@ -23,7 +23,7 @@
 const Sentry = __importStar(require("@sentry/browser"));
 const logs_1 = require("../utils/logs");
 function showAuthorizationError(e) {
-    const dialog = require('@electron/remote').dialog;
+    const dialog = require('@electron/remote/main').dialog;
     if (dialog) {
         dialog
             .showMessageBox({
@@ -43,6 +43,7 @@
         });
     }
     else {
+        console.log(e);
         logs_1.authLog.error('Unable to show authorization error: ', { e });
     }
 }
