--- src/original/moba-desktop/src/utils/desktop-bridge.js
+++ src/modified/moba-desktop/src/utils/desktop-bridge.js
@@ -32,8 +32,9 @@
 exports.bridge = {
     version: remote_1.app.getVersion() === '0.0.1' ? version_const_1.DEV_DESKTOP_APP_VERSION : remote_1.app.getVersion(),
     platform,
-    isWin: platform !== 'darwin',
+    isWin: platform === 'win32',
     isMac: platform === 'darwin',
+    isLinux: platform === 'linux',
     logsPath: path.resolve(remote_1.app.getPath('logs')),
     settings: null,
     settingsV2: null,
