--- src/original/moba-desktop/src/auth/auth-flow.js
+++ src/modified/moba-desktop/src/auth/auth-flow.js
@@ -60,7 +60,7 @@
             return authFlowData;
         }
         else {
-            throw Error(`Auth flow data doesn't match: ${code}, ${state}, ${scope}, ${JSON.stringify(authFlowData)}`);
+            throw Error(`Auth flow data doesn't match: ${code}, ${state}, ${scope}, Data: ${JSON.stringify(authFlowData)}`);
         }
     })
         .then(authFlowData => receiveToken(code, authFlowData.verifier, state))
