--- src/original/moba-desktop/src/utils/game-overlay/game-overlay-window.js
+++ src/modified/moba-desktop/src/utils/game-overlay/game-overlay-window.js
@@ -1,7 +1,7 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.onOverlayWindowContentLoaded = void 0;
-const remote_1 = require("@electron/remote");
+const remote_1 = require("@electron/remote/main");
 const isOverlayPage = (content) => {
     return /mobalytics-in-game-overlay-marker/gim.test(content);
 };
